apps/prairielearn/src/models/questions.ts

```diff
@@ -0,0 +1,54 @@
+import sqldb = require('@prairielearn/postgres');
+import AnsiUp from 'ansi_up';
+import {
+  CourseInstance,
+  TopicSchema,
+  TagsForQuestionSchema,
+  AssessmentsFormatForQuestionSchema,
+} from '../lib/db-types';
+import { z } from 'zod';
+
+const QuestionsTableDataSchema = z.object({
+  id: z.string(),
+  qid: z.string(),
+  title: z.string(),
+  sync_errors: z.string().nullable(),
+  sync_warnings: z.string().nullable(),
+  grading_method: z.string(),
+  external_grading_image: z.string().nullable(),
+  display_type: z.string(),
+  open_issue_count: z.string(),
+  topic: TopicSchema,
+  tags: TagsForQuestionSchema,
+  assessments: AssessmentsFormatForQuestionSchema,
+});
+export type QuestionsTableData = z.infer<typeof QuestionsTableDataSchema>;
+
+const ansiUp = new AnsiUp();
+const sql = sqldb.loadSqlEquiv(__filename);
+
+export async function selectQuestionsForCourse(
+  course_id: string | number,
+  course_instances: CourseInstance[],
+): Promise<QuestionsTableData[]> {
+  const rows = await sqldb.queryRows(
+    sql.select_questions_for_course,
+    {
+      course_id: course_id,
+    },
+    QuestionsTableDataSchema,
+  );
+
+  const ci_ids = course_instances.map((ci) => ci.id);
+  const questions = rows.map((row) => {
+    if (row.sync_errors) row.sync_errors_ansified = ansiUp.ansi_to_html(row.sync_errors);
+    if (row.sync_warnings) {
+      row.sync_warnings_ansified = ansiUp.ansi_to_html(row.sync_warnings);
+    }
+    row.assessments = row.assessments?.filter((assessment) =>
+      ci_ids.includes(assessment.course_instance_id),
+    );
+    return row;
+  });
```

Comment:
now that the query result is typed, the compiler is pretty unhappy about these '*_ansified' things being added. 

What is your preferred method to deal with this? It seems like these are at least a few places in the PL codebase so we probably want some consistent thing to do.

Reply:
```suggestion
  const questions = rows.map((row) => ({
      ...row,
      sync_errors_ansified: row.sync_errors && ansiUp.ansi_to_html(row.sync_errors),
      sync_warnings_ansified: row.sync_warnings && ansiUp.ansi_to_html(row.sync_warnings),
      assessments: row.assessments?.filter((assessment) =>
        ci_ids.includes(assessment.course_instance_id),
      ),
  }));
```

This would require two different types, one for the data from the SQL, one from the return value of the function. The latter can extend the former.

Reply:
Ya, I was thinking something similar. Looks good to me as long as @mwest1066 and @nwalters512 are cool with it.

Reply:
done